addon-test-support/ember-qunit/-internal/test-debug-info-summary.js

```diff
@@ -0,0 +1,149 @@
+const PENDING_REQUESTS = 'Pending AJAX requests';
+const PENDING_TEST_WAITERS = 'Pending test waiters: YES';
+const PENDING_TIMERS = 'Pending timers';
+const PENDING_SCHEDULED_ITEMS = 'Pending scheduled items';
+const ACTIVE_RUNLOOPS = 'Active runloops: YES';
+
+export function getSummary(testCounts, leakCounts, testNames) {
+  let leakInfo =
+    leakCounts.length > 0
+      ? `We found the following information that may help you identify code that violated test isolation: \n
+  ${leakCounts.join('\n')}
+   \n`
+      : '';
+  let summary = `TESTS ARE NOT ISOLATED\n
+ The following ${testCounts} test(s) have one or more issues that are resulting in non-isolation (async execution is extending beyond the duration of the test):\n
+ ${testNames.join('\n')}
+\n
+ ${leakInfo}
+ More information has been printed to the console. Please use that information to help in debugging.
+`;
+
+  return summary;
+}
+
+export default class TestDebugInfoSummary {
+  constructor() {
+    this.reset();
+  }
+
+  add(testDebugInfo) {
+    let summary = testDebugInfo.summary;
+
+    this._testDebugInfos.push(testDebugInfo);
+
+    this.fullTestNames.push(summary.fullTestName);
+    if (summary.hasPendingRequests) {
+      this.hasPendingRequests = true;
+    }
+    if (summary.hasPendingWaiters) {
+      this.hasPendingWaiters = true;
+    }
+    if (summary.hasRunLoop) {
+      this.hasRunLoop = true;
+    }
+    if (summary.hasPendingTimers) {
+      this.hasPendingTimers = true;
+    }
+    if (summary.pendingScheduledQueueItemCount > 0) {
+      this.hasPendingScheduledQueueItems = true;
+    }
+
+    this.totalPendingRequestCount += summary.pendingRequestCount;
+    this.totalPendingTimersCount += summary.pendingTimersCount;
+    this.totalPendingScheduledQueueItemCount += summary.pendingScheduledQueueItemCount;
+  }
+
+  get hasDebugInfo() {
+    return this._testDebugInfos.length > 0;
+  }
+
+  reset() {
+    this._testDebugInfos = [];
+    this.fullTestNames = [];
+    this.hasPendingRequests = false;
+    this.hasPendingWaiters = false;
+    this.hasRunLoop = false;
+    this.hasPendingTimers = false;
+    this.hasPendingScheduledQueueItems = false;
+    this.totalPendingRequestCount = 0;
+    this.totalPendingTimersCount = 0;
+    this.totalPendingScheduledQueueItemCount = 0;
+  }
+
+  printToConsole(_console = console) {
```

Comment:
Does `printToConsole(console = console) {` work? I wasn't sure RE: shadowing, but it might make highlighting nicer..

Reply:
I had that first, but found it was extremely trolly to see that. Seeing `console = console` is very confusing, IMHO. Thoughts?

Reply:
Ya I'm going to leave this as is for now.

Reply:
Agree. Thank you 😊